src/main/java/com/powsybl/openloadflow/sa/LimitViolationManager.java

```diff
@@ -180,16 +176,25 @@ private void detectBusViolations(LfBus bus) {
         double busV = bus.getV();
         if (!Double.isNaN(bus.getHighVoltageLimit()) && busV > bus.getHighVoltageLimit()) {
             LimitViolation limitViolation1 = new LimitViolation(bus.getVoltageLevelId(), LimitViolationType.HIGH_VOLTAGE, bus.getHighVoltageLimit() * scale,
-                    (float) 1., busV * scale);
+                    (float) 1., busV * scale, createViolationLocation(bus));
             addBusLimitViolation(limitViolation1, bus);
         }
         if (!Double.isNaN(bus.getLowVoltageLimit()) && busV < bus.getLowVoltageLimit()) {
             LimitViolation limitViolation2 = new LimitViolation(bus.getVoltageLevelId(), LimitViolationType.LOW_VOLTAGE, bus.getLowVoltageLimit() * scale,
-                    (float) 1., busV * scale);
+                    (float) 1., busV * scale, createViolationLocation(bus));
             addBusLimitViolation(limitViolation2, bus);
         }
     }
 
+    public static ViolationLocation createViolationLocation(LfBus bus) {
+        List<Integer> nodes = bus.getNodes();
+        if (nodes.isEmpty()) {
+            return null;
+        }
+
+        return new NodeBreakerViolationLocation(bus.getVoltageLevelId(), nodes);
+    }
+
     /**
      * Detect violation limits on one voltage angle limit and add them to the given list
      * @param limit voltage angle limit of interest
```

Comment:
Can you add a test in NodeBreaker and BusBreaker that verifies that the results are as expected ?